docs/csharp/programming-guide/concepts/async/async-return-types.md

```diff
@@ -1,63 +1,74 @@
+- <xref:Windows.Foundation.IAsyncOperation%601>, for async operations in UWP that return a value.
+- <xref:Windows.Foundation.IAsyncOperationWithProgress%602>, for async operations in UWP that report progress and return a value.
+
+## Task return type
+
+Async methods that don't contain a `return` statement or that contain a `return` statement that doesn't return an operand usually have a return type of <xref:System.Threading.Tasks.Task>. Such methods return `void` if they run synchronously. If you use a <xref:System.Threading.Tasks.Task> return type for an async method, a calling method can use an `await` operator to suspend the caller's completion until the called async method has finished.
+
+In the following example, the `WaitAndApologizeAsync` method doesn't contain a `return` statement, so the method returns a <xref:System.Threading.Tasks.Task> object. Returning a `Task` enables `WaitAndApologizeAsync` to be awaited. The <xref:System.Threading.Tasks.Task> type doesn't include a `Result` property because it has no return value.
+
+:::code language="csharp" source="async-return-types/async-returns2.cs" id="TaskReturn":::
+
+`WaitAndApologizeAsync` is awaited by using an await statement instead of an await expression, similar to the calling statement for a synchronous void-returning method. The application of an await operator in this case doesn't produce a value.
 
-:::code language="csharp" source="./snippets/async-return-types/async-returns1a.cs" id="SnippetSecondVersion":::
+You can separate the call to `WaitAndApologizeAsync` from the application of an await operator, as the following code shows. However, remember that a `Task` doesn't have a `Result` property, and that no value is produced when an await operator is applied to a `Task`.
 
-## Task Return Type  
-Async methods that don't contain a `return` statement or that contain a `return` statement that doesn't return an operand usually have a return type of <xref:System.Threading.Tasks.Task>. Such methods return `void` if they run synchronously. If you use a <xref:System.Threading.Tasks.Task> return type for an async method, a calling method can use an `await` operator to suspend the caller's completion until the called async method has finished.  
-  
-In the following example, the `WaitAndApologize` async method doesn't contain a `return` statement, so the method returns a <xref:System.Threading.Tasks.Task> object. Returning a `Task` enables `WaitAndApologize` to be awaited. The <xref:System.Threading.Tasks.Task> type doesn't include a `Result` property because it has no return value.  
+The following code separates calling the `WaitAndApologizeAsync` method from awaiting the task that the method returns.
 
-:::code language="csharp" source="./snippets/async-return-types/async-returns2.cs" id="SnippetTaskReturn":::
+:::code language="csharp" source="async-return-types/async-returns2a.cs" id="AwaitTask":::
```

Comment:
I think the "snippets" part of the path is removed by mistake, here and in other snippets too.

Reply:
I'm just now seeing many review comments, this is a draft PR. I am still actively working on this and there are many changes yet to be made. I have already fixed this, and several other things locally. It might be better to wait until this is promoted to an actual PR before spending too much time on the draft now.